Book.cpp

```diff
@@ -82,35 +84,43 @@ bool leapyear(int y) {
 
 std::ostream& operator<<(std::ostream& os, const Date& d)
 {
-	//gestiamo qua questo if o dopo?
-	/* Forse meglio dopo
 	if (!d.exist())
-		return;
-	*/
+		return os << "";
 	return os << d.day() << '/' << d.month() << '/' << d.year();
 }
 
 //IMPLEMENTAZIONE BOOK -----------------------------
 //costruttori ...
 
+Book::Book(std::string name, std::string surname, std::string title, std::string ISBN, int day = DefaultDay, Month month = DefaultMonth, int year = DefaultYear, bool checkout = DefaultCheckout)
+	: name_ {name}, surname_ {surname}, title_ {title}, ISBN_ {ISBN}, day_ {day}, month_ {month}, year_ {year}, checkout_ {checkout}
+{
+}
+
 void Book::lent(){
 	checkout_ = true;
 }
 void Book::restituted() {
 	checkout_ = false;
 }
 
+bool Book::exist_date(void){
+	return month_
+}
 bool operator==(Book a, Book b) {
 	return a.ISBN() == b.ISBN();
 }
 bool operator!=(Book a, Book b) {
 	return !(a == b);
 }
 std::ostream& operator<<(std::ostream& os, Book a) {
-	os << title_ << '\n' <<
-				name << ' ' << surname << '\n' <<
-				ISBN_ << '\n';
-	if (exist_)
-		os << copyright_ << '\n';
-	return os;
+	std::string t = a.title() + "\n" + a.name() + " " + a.surname();
+	if(a.valid()){
+		t = t + "\n" + a.day() + " " + a.month() + " " + a.year();
+	}
+  if(!a.checkout()){
+    t = t + "\nNON ";
+  }
+  t = t + "IN PRESTITO";
```

Comment:
Se non entriamo nell'if, "IN PRESTITO" sarà attaccato all'anno del copyright, o al cognome; potremmo fare così:
`t = t + "\n";
 if(!a.checkout()){
    t = t + "NON ";
  }
  t = t + "IN PRESTITO";`